internal/connector/internal/handlers/connector_admin.go

```diff
@@ -400,6 +404,97 @@ func (h *ConnectorAdminHandler) GetConnector(writer http.ResponseWriter, request
 	handlers.HandleGet(writer, request, &cfg)
 }
 
+func (h *ConnectorAdminHandler) PatchConnector(writer http.ResponseWriter, request *http.Request) {
+	connectorId := mux.Vars(request)["connector_id"]
+	contentType := request.Header.Get("Content-Type")
+
+	cfg := &handlers.HandlerConfig{
+		Validate: []handlers.Validate{
+			handlers.Validation("connector_id", &connectorId, handlers.MinLen(1), handlers.MaxLen(maxConnectorIdLength)),
+			handlers.Validation("Content-Type header", &contentType, handlers.IsOneOf("application/json", "application/json-patch+json", "application/merge-patch+json")),
+		},
+		Action: func() (i interface{}, serviceError *errors.ServiceError) {
```

Comment:
As @dhirajsb correctly mentioned, this code is a duplication of https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/db5cd9dffce3baeb27450cbdd819ccabfd558ce4/internal/connector/internal/handlers/connectors.go#L135 the common code must be extracted and reused among the two handlers, otherwise it will become unmaintainable.